tests/integration/test_charm.py

```diff
@@ -1,26 +1,159 @@
+    """Helper function to wait for runner instances to be ready.
+
+    Args:
+      app: Application instance to check the runners.
+      num: Number of runner instances to check for.
+    """
+    action = await app.units[0].run("lxc list --format json")
+    await action.wait()
+
+    assert action.results["return-code"] == 0
+
+    lxc_instance = json.loads(action.results["stdout"])
+    assert len(lxc_instance) == num
+
+    for instance in lxc_instance:
+        action = await app.units[0].run(f"lxc exec {instance['name']} -- ps aux")
+        await action.wait()
+
+        assert f"/bin/bash {Runner.runner_script}" in action.results["stdout"]
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.dependency(depends=["test_check_runners_with_no_runner"])
+async def test_reconcile_runners_spawn_one(model: Model, app: Application) -> None:
+    """
+    arrange: An working application of github-runner with no runners.
+    act: Set number of runner to 1, and run reconcile-runners.
+    assert: One runner instance exists.
+    """
+    await app.set_config({"virtual-machines": "1"})
+    action = await app.units[0].run_action("reconcile-runners")
+    await action.wait()
+    await model.wait_for_idle()
+
+    await check_runner_instance(app, 1)
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.dependency(depends=["test_reconcile_runners_spawn_one"])
+async def test_check_runners(model: Model, app: Application, app_name: str) -> None:
+    """
+    arrange: An working application of github-runner with one runner.
+    act: Run check-runners action.
+    assert: Action returns with one runner.
+    """
+    await model.wait_for_idle()
+    action = await app.units[0].run_action("check-runners")
+    await action.wait()
+
+    assert action.results["online"] == "1"
+    assert action.results["offline"] == "0"
+    assert action.results["unknown"] == "0"
+
+    runner_names = action.results["runners"].split(", ")
+    assert len(runner_names) == 1
+    assert runner_names[0].startswith(f"{app_name}-0")
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+@pytest.mark.dependency(depends=["test_check_runners"])
+async def test_change_token(model: Model, app: Application, token_two: str) -> None:
+    """
+    arrange: An working application of github-runner with one runner.
+    act: Change the token in charm configuration.
+    assert: Existing runners are flushed, and repo-policy-compliance is using the new token.
+    """
+    await app.set_config({"token": token_two})
+    await model.wait_for_idle()
+
+    # The old runner should be flushed out.
+    action = await app.units[0].run_action("check-runners")
+    await action.wait()
+
+    assert action.results["online"] == "0"
+    assert action.results["offline"] == "0"
+    assert action.results["unknown"] == "0"
 
+    action = await app.units[0].run("cat /etc/systemd/system/repo-policy-compliance.service")
```

Comment:
I was about to comment that recently @jdkandersson added a review on a Synapse PR about checking if the action succeed besides the result but then I realized that this is being done here, right? I was just confused by the name of the variable "action".

Reply:
I added checking of `action.status == "completed"` to each instance of `juju.action.Action` object usage.

Reply:
The point is more that we should get a second opinion on whether the action succeeded, e.g., in this case check that there are no more lxd machines or something like that